site/source/docs/contracts_configuration.md

```diff
@@ -186,6 +186,30 @@ contracts: {
 ...
 ```
 
+## Dynamic Addresses
+
+There are scenarios in which we want to configure a Smart Contract that is already deployed by a third-party, but its address can only be computed at run-time. For such cases, Embark supports specifying an `addressHandler` which returns the address we're interested in. Usually, other Smart Contract instances are needed to perform that computation, so the [`deps` configuration](#Deployment-hooks) comes in handy as well:
+
+```
+contracts: {
+  SimpleStorage: {
+    fromIndex: 0,
+    args: [100],
+  },
+  OtherContract: {
+    deps: ['SimpleStorage'],
+    addressHandler: async (deps) => {
+      // use `deps.contracts.SimpleStorage` to determine address
+    },
+    abiDefinition: ABI
+  },
+}
+```
+
+In the example above, `OtherContract` will be deployed after `SimpleStorage` because it uses the `deps` property to define Smart Contracts that it depends on. All dependencies are exposed on the `addressHandler` function parameter similar to [deployment hooks](#Deployment-hooks).
+
+**Note**: This only works when an ABI is provided for the Smart Contract thatis already deployed (in this case `OtherContract`).
```

Comment:
> for the Smart Contract thatis already deployed

"thatis" needs to be split into "that is", but mainly I find that sentence confusing because in the preceding text it says:

> `OtherContract` will be deployed after `SimpleStorage`

I'm making an inexpert observation, i.e. contracts and contract deployment are two aspects of Embark that I don't understand as well as others.

Reply:
I would just remove the Note, or change it. It is not true that it only works with an `abiDefinition`. The `abiDefinition` is only needed if the contract is not on disk.

Reply:
> The abiDefinition is only needed if the contract is not on disk.

That is technically correct. Will update the message. I was thinking that this feature really only makes sense to use if you don't actually own the source of the contract, cause if you do, you might as well just deploy it and don't need a function to compute it at run-time. So having an ABI would be sufficient.

But then I realized there's cases where owned contracts are also already deployed (using different environments), in which case the source is available.